### PR TITLE
refactor: Move db_session and embedding_model args into AppConfig

### PR DESCRIPTION
**Pull request by yoomlam**
## Ticket

From [Slack](https://nava.slack.com/archives/C06DP498D1D/p1721764756834649):

> - [retrieve_with_scores()](https://github.com/navapbc/labs-decision-support-tool/blob/cb9d663d1f575cc06119f561893eff3ee099288e/app/src/retrieve.py#L15) and [_ingest_cards()](https://github.com/navapbc/labs-decision-support-tool/blob/cb9d663d1f575cc06119f561893eff3ee099288e/app/src/ingest_guru_cards.py#L26) take the 2 arguments `db_session` and `embedding_model`.
> - I see these 2 arguments as "[context](https://www.baeldung.com/cs/context-design-pattern)" or "fixtures", and not really arguments that would change with different calls to these functions.
> - Additionally, any retrieval function should use the same `embedding_model` as the ingest function.
> What do y'all think about wrapping these 2 arguments into a context class or moving them into the `AppConfig` class?

## Changes

Move `db_session` and `embedding_model` args into `AppConfig` to be used as a context object.

## Testing

No changes in software behavior.

### CODE REVIEW

File: app/tests/src/test_ingest_guru_cards.py
Comment (yoomlam) [inline app/tests/src/test_ingest_guru_cards.py:49]: Include the `app_config` fixture to mock the `sentence_transformer` embedding model used by `_ingest_cards()`.

File: app/tests/src/test_retrieve.py
Comment (yoomlam) [inline app/tests/src/test_retrieve.py:9]: `mock_embedding_model`/`MockSentenceTransformer` is now provided by the `app_config` fixture via `MockAppConfig.sentence_transformer()`.

File: app/src/app_config.py
Comment (yoomlam) [inline app/src/app_config.py:33]: `AppConfig` functions as a context object that provides access to the DB and `sentence_transformer` embedding model instance.

File: app/src/retrieve.py
Comment (yoomlam) [inline app/src/retrieve.py:15]: Remove context arguments, which are now replaced with fields from `get_app_config()`

File: app/src/shared.py
Comment (KevinJBoyer) [reply app/src/shared.py:15]: At this point, should we also move get_app_config to app_config.py and delete this file?

File: app/src/ingest_guru_cards.py
Comment (yoomlam) [reply app/src/ingest_guru_cards.py:78]: I started to do that but noticed the explicit `db_session.commit()`, so I unsure if we wanted to keep the DB transaction explicit.

Comment (KevinJBoyer) [reply app/src/ingest_guru_cards.py:78]: That's a good point -- defer to you on what you think makes the most sense!

File: app/src/retrieve.py
Comment (ccheng26) [inline app/src/retrieve.py:19]: nit: does get_app_config() need to be called everytime or can we store the results in a variable?

File: app/src/ingest_guru_cards.py
Comment (yoomlam) [reply app/src/ingest_guru_cards.py:78]: I defer this to a different PR.

File: app/src/retrieve.py
Comment (yoomlam) [reply app/src/retrieve.py:19]: Good point!

Comment (yoomlam) [reply app/src/retrieve.py]: I had to make more changes that anticipated, especially around login, so test the login mechanism carefully and check the deployment after this merges.

File: app/src/app_config.py
Comment (yoomlam) [inline app/src/app_config.py:37]: Assuming `AppConfig` only ever needs to be instantiated once per application startup.

File: app/src/login.py
Comment (yoomlam) [inline app/src/login.py:14]: Clarified wording. `GLOBAL_PASSWORD` is not needed by Chainlit.

File: app/tests/src/test_login.py
Comment (yoomlam) [inline app/tests/src/test_login.py:7]: Set the `global_password` directly and avoid having to `cache_clear()`. This assumes that `PydanticBaseEnvConfig` correctly sets `global_password` from the env variable, which is a valid assumption since the point of this test is `login.py`.

**Review by ccheng26 — APPROVED** [review]
code ran without issues, lgtm

**Pull request merged**
